Using paddle-python for API calls

pyepye opened the pull request:
* I created https://github.com/pyepye/paddle-python so I can talk to Paddle outside of Django. It's well tested and has type hints so I thought it would be helpful to use instead. Especially as we use Paddle's API more.
* Fix model string names - `<Plan: Plan <Plan 10:10>>` is now `<Plan: Plan 10:10>`
* I tried to make models.py a little more Pythonic (Moved the Price model down the file a little, no actual changes)
* Cleaned up the docs to give some better advice around `Post-Checkout` with some better examples using paddle-python
* Check the event time when updating the Subscription model to ensure only the latest events are sync to the database

FPurchess commented:
@pyepye again, thanks for your work! Before merging, I'd love to see the coverage back up again so the build won't break.

pyepye commented on djpaddle/models.py:
```diff
@@ -57,45 +47,52 @@ class Meta:
+        recurring_price = data.pop("recurring_price", {})
 
-        plan, _ = cls.objects.get_or_create(pk=pk, defaults=data)
+        plan, __ = cls.objects.get_or_create(pk=pk, defaults=data)
```
It's because the python interpreter stores the last expression value to the special variable called `_`. 
As `_` is technically already a variable it can be dangerous to use and can lead to some odd behaviour such as:
```
>>> 10
10
>>> _
10
>>> t, _ = [1, 2]
>>> _
2
>>> 10
10
>>> _
2
````

Using a double underscore on the other hand isn't a special variable so it's less likely it will be misused but is read the same.

You could take this to the extreme though and do something like:
(but that's so verbose it think it's a little confusing.)
```
plan, created = cls.objects.get_or_create(pk=pk, defaults=data)
del created
```

In reality it doesn't really matter though and the change was just force of habit. I'm happy to turn it back if you would like.

pyepye commented on djpaddle/models.py:
```diff
@@ -189,11 +184,20 @@ def _sanitize_webhook_payload(cls, payload):
+        except cls.DoesNotExist:
+            return cls.objects.create(pk=pk, **data)
+
+        event_time = datetime.strptime(data["event_time"], "%Y-%m-%d %H:%M:%S")
```
👍 Good shout

pyepye commented on djpaddle/settings.py:
```diff
@@ -12,11 +14,12 @@
 if not DJPADDLE_VENDOR_ID:
     raise ImproperlyConfigured("'DJPADDLE_VENDOR_ID' must be set")
-
+os.environ["PADDLE_VENDOR_ID"] = DJPADDLE_VENDOR_ID
```
Yea totally agree, it was terrible idea.

pyepye commented:
I've replaced:
```
try:
    data["plan"] = Plan.objects.get(pk=plan_id)
except Plan.DoesNotExist:
    data["plan"] = None
```
As when I removed it and wrote a test without a Plan I got the following error:
```
django.db.utils.IntegrityError: NOT NULL constraint failed: djpaddle_subscription.plan_id
```
Now it actually goes off to Paddle and adds the plan.

pyepye commented on djpaddle/models.py:
```diff
@@ -1,11 +1,17 @@
+from . import settings, signals
 from .fields import PaddleCurrencyCodeField
 
+PADDLE_CLIENT = Paddle()
```
I went for caps as I felt it was a "global enough" but it doesn't "feel" great when using it within functions so I've lowered it.

As for the name. I spend a while thinking about the choice between `PaddleClient`, `PaddleAPI` (PaddleApi`) and just `Paddle`). I had a look at quite a few other clients I had used and everyone did it in different ways so I just went for the shortest.
You saying that shows I perhaps didn't make the correct choice, although I think it's the most confusing in this repo which is all about Paddle, In others it isn't quite as confusing.
